lib/modules.nix

```diff
@@ -127,10 +127,11 @@ let
       # When extended with extendModules or moduleType, a fresh instance of
       # this module is used, to avoid conflicts and allow chaining of
       # extendModules.
-      internalModule = rec {
+      # Cannot use 'config' here. This causes infinite recursion.
```

Comment:
This statement is too broad.
Depending on how we define "here", it's either always true everywhere: a module like `{ config, ... }: builtins.seq config { ... }` never works.
If you mean anything else you'd have to be precise about it. It's probably fine to refer to `config` in an option `description`, for example.
I general I think it's tempting to put comments describing discoveries about strictness all over the place, but they're tricky to get right, and usually not all that relevant.

```suggestion
```